src/cljs/rems/user_settings.cljs

```diff
@@ -81,10 +81,30 @@
                    (when first-time? ; to avoid an infinite loop if the settings fail to save (e.g. unsupported language in cookies)
                      [[:check-if-should-save-language!]]))})))
 
-(defn fetch-user-settings! []
+(rf/reg-event-fx
+ :flash-message-error
+ (fn [_ user-settings-error]
+   ;; TODO: refactor this part
+   ;; this doesnt work
+   ;; (flash-message/default-error-handler :top "Fetch user settings")
+   ;; this works
+   (flash-message/show-default-error! :top (str "Fetch user settings" "wtf"))
+   {}))
+
+(rf/reg-event-fx
+ :loaded-user-settings-fail
+ (fn [{:keys [db]} [_ user-settings-error]]
+   (let [logged-in? (:user (:identity db))]
+     (if-not logged-in?
+       {:db db}
+       {:db db
+        :dispatch [:flash-message-error user-settings-error]}))))
+
+(defn fetch-user-settings! [opts]
   (fetch "/api/user-settings"
-         {:handler #(rf/dispatch-sync [:loaded-user-settings %])
-          :error-handler (flash-message/default-error-handler :top "Fetch user settings")}))
+         (merge opts
+                {:handler #(rf/dispatch-sync [:loaded-user-settings %])}
+                {:error-handler #(rf/dispatch [:loaded-user-settings-fail %])})))
```

Comment:
You could do a simple check like `(if (= 401 (:status response)) ... ...)` here.